src/Pages/Login.tsx

```diff
@@ -60,29 +51,38 @@ export const Login = () => {
     navigate("/");
   };
 
+  const { setToken } = useToken();
+
   const onSubmit = async (data: FormValues) => {
     await axios
       .post("http://localhost:4000/login", data)
-      .then(() => {
-        // setLogggedIn(true)
-        showSuccessAlert("logged in");
-        navigate("/dashboard");
-        //navigate("/");
+      .then((response) => {
+        const user = response.data.user;
+        const token = response.data.token;
+
+        setToken(token); //The token from the response is saved using useToken. ustawiam Token na ten otrzymany z bazy
+        setUser(user); //setUser updates the AuthContext with the logged-in user’s details.
+        localStorage.setItem("userId", user.id); // After a successful login, the user's id is saved in localStorage
+        localStorage.setItem("userRole", user.role);
+        showSuccessAlert("Logged in successfully");
+
+        if (user.role === "ADMIN") {
+          navigate("/dashboard");
+        } else {
+          navigate("/");
+        }
       })
       .catch((err) => {
-        // if (err.response && err.response.data) { //skoro to jest async/await to czemu nie robimy tegow try/catch
-        //   setLoginError(err.response.data);
-        // } else {
-        //   setLoginError("Unexpected error");
-        // }
-        // setLoginError(err.response.data ?? 'Unexpected error') -> krotsza wersja przykładu wyzej
-        showErrorAlert((err as unknown as Error).message ?? "Unexpected error");
-        /*
-        setLoginError(err.response.data ? err.response.data : 'Unexpected error')
-        */
+        showErrorAlert(
+          (err.response?.data?.message as string) ?? "Unexpected error"
```

Comment:
mozna zmienic metode `showErrorAlert`, zeby akceptowala `string | undefined` i warunek, ktory zrobiles tutaj przeniesc do metody. Wtedy nie bedziesz potrzebowal tego warunku w kazdym miejscu ( a bedzie sie to powtarzac)